packages/nextjs/components/Header.tsx

```diff
@@ -1,17 +1,53 @@
-import React from "react";
-import { useRouter } from "next/router";
+import React, { useState, useRef, useEffect } from "react";
 import Link from "next/link";
 import Image from "next/image";
 import { Faucet } from "~~/components/scaffold-eth";
 import RainbowKitCustomConnectButton from "~~/components/scaffold-eth/RainbowKitCustomConnectButton";
-import { BugAntIcon, SparklesIcon } from "@heroicons/react/24/outline";
+import { Bars3Icon, BugAntIcon, SparklesIcon } from "@heroicons/react/24/outline";
+import { useRouter } from "next/router";
+
+/**
+ * Defs for NavLinks
+ ** @note Are we sure we don't want this in a separate component?
+
+ ** @note I looked for a method to pull the routes from next useRouter, is it possible to auto-populate
+          this list?
```

Comment:
> Are we sure we don't want this in a separate component?

Yup, as mentioned in the previous comment since `NavLink` component is small and will not be used by any other component we can keep it here : )

> I looked for a method to pull the routes from next useRouter, is it possible to auto-populate
    this list?

The returned [router object](https://nextjs.org/docs/api-reference/next/router#router-object) from `useRouter` don't seem to have any direct property/method to access all routes, but I might have missed something. 
Also, the current approach  looks good to me since we want to give options to add custom Icons too. 

PS : we can remove this comments : )